config/connector.go

```diff
@@ -0,0 +1,74 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config // import "go.opentelemetry.io/collector/config"
+import (
+	"go.opentelemetry.io/collector/confmap"
+)
+
+// Connector is the configuration of a component.Connector. Specific extensions must implement
+// this interface and must embed ConnectorSettings struct or a struct that extends it.
+type Connector interface {
+	identifiable
+	validatable
+
+	// Implement both to ensure:
+	// 1. Only connectors are defined in 'connectors' section
+	// 2. Connectors may be placed in receiver and exporter positions in pipelines
```

Comment:
Do we want 2.? It may confuse users.

Reply:
Hmm, this is fundamental to the design as I understood it.

The idea was that you _must_ use a connector as both a receiver and an exporter. For example:

```yaml
logs/in:
  receivers: [filelog]
  exporters: [countconnector]

metrics/out:
  receivers: [countconnector]
  exporters: [metricsexporter]
```

I think this can get confusing in very complex cases, but the basic idea is simple enough to start with. Are you thinking something else in particular, such as having dedicated connector sections in pipelines?

Reply:
Ah, that makes sense. I had not considered this - will take a closer look and figure out how to disallow this.

Reply:
I had the wrong idea about how the config enforced component usage. Fortunately I think this was an easy fix.